drivers/virtio/VirtioBlk.c

```
@@ -214,141 +229,169 @@ void InitializeVirtioBlk(PciDevice device) {
         pending_reqs[i].status = NULL;
     }
 
-    PrintKernel("VirtIO-Blk: Driver initialized successfully!\n");
+    // 8. Set DRIVER_OK status bit
+    common_cfg_ptr->device_status |= (1 << 2);
+    
+    PrintKernelSuccess("VirtIO-Blk: Device initialized successfully\n");
+    
+    // Register as block device
+    uint64_t total_sectors = 0x1000000; // Default 8GB, should read from device config
+    const char* dev_name = GenerateDriveName(DEVICE_TYPE_VIRTIO);
```

Clang (14.0.6)

[warning] variable 'dev_name' is not initialized

(cppcoreguidelines-init-variables)